src/cage/src/memory/shared.rs

```diff
@@ -148,16 +148,24 @@ impl ShmSegment {
             }
         };
 
-        unsafe {
-            (libc::mmap(
+        let result = unsafe {
+            libc::mmap(
                 shmaddr as *mut c_void,
                 self.size as usize,
                 prot,
                 (MAP_SHARED as i32) | (MAP_FIXED as i32),
                 fobjfdno,
                 0,
-            ) as usize)
+            ) as usize
+        };
+
+        // Check if mmap failed and return the appropriate error
+        if result as isize == -1 {
+            let errno = get_errno();
+            return handle_errno(errno, "map_shm") as usize;
         }
+
+        result
```

Comment:
we also need to return the result below in unmap_shm

likewise, we need to make sure the errors are also properly returned from the calling functions shmat_helper and shmdt_helper